### PR TITLE
[STK-510] Next js and basic setup

### PR DESCRIPTION
**Pull request by wixzi**
Next js and basic setup

### CODE REVIEW

**Comment by berteotti** [comment]
I've some questions about the initial setup.

- Should we use `/appDir`?
- Do we need the `ui` package? If we are creating other packages than `app` (like landing page), I can see a big reason for it.
- The same think for `ui` applies to `utils`.

Otherwise, good job Velu!

**Comment by Diogomartf** [comment]
> I've some questions about the initial setup.
> 
> * Should we use nextjs `/app`?
> * Do we need the `ui` package? If we are creating other packages than `app` (like landing page), I can see a big reason for it.
> * The same think for `ui` applies to `utils`.
> 
> Otherwise, good job Velu!

Let's use the app router and the new file system.

https://nextjs.org/docs/app/building-your-application/routing

image

File: packages/app/next.config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  transpilePackages: ["@stackly/ui"],
```
Comment (Diogomartf) [inline packages/app/next.config.js:3]: Not sure if we need the UI outside of app at this point.

The only potential use for the UI lib is if we build the landing page on other repo and want to use some components.

File: packages/app/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "start": "next start",
+    "lint": "next lint",
+    "export": "next export",
+    "typecheck": "tsc --noEmit",
+    "clean": "rm -rf .vercel && rm -rf .turbo && rm -rf out && rm -rf .next"
+  },
+  "dependencies": {
+    "@stackly/ui": "*",
+    "@stackly/utils": "*",
+    "next": "canary",
```
Comment (Diogomartf) [inline packages/app/package.json:19]: I think we should specify a version, like the latest 13.4 

https://nextjs.org/blog/next-13-4

**Comment by Diogomartf** [comment]
> * Should we use nextjs `/app`?

I do think so, we should use the new app router. I've been using it to rebuild my website and has been great. And I think it will be even better as a team because there are a lot of good conventions.

> * Do we need the `ui` package? If we are creating other packages than `app` (like landing page), I can see a big reason for it.
> * The same think for `ui` applies to `utils`.

The question here is: Can we forecast the use the `ui` lib in other repos? If not we can move the `ui` to inside `app`.
Same with utils, will we share utils across repos or packages?

File: packages/app/next.config.js (hunk 1 above)
Comment (wixzi) [reply packages/app/next.config.js:3]: Moved inside the app

File: packages/app/package.json (hunk 2 above)
Comment (wixzi) [reply packages/app/package.json]: Added proper version

**Review by berteotti — APPROVED** [review]
Looks awesome, good job Velu!

**Pull request merged**